nuxeo-core-binarymanager-s3/src/test/java/org/nuxeo/ecm/core/storage/sql/TestS3BinaryManager2WithSSE2AndPolicy.java

```diff
@@ -0,0 +1,45 @@
+package org.nuxeo.ecm.core.storage.sql;
+
+import static org.junit.Assume.assumeTrue;
+import static org.nuxeo.ecm.core.storage.sql.S3BinaryManager.AWS_ID_PROPERTY;
+import static org.nuxeo.ecm.core.storage.sql.S3BinaryManager.AWS_SECRET_PROPERTY;
+import static org.nuxeo.ecm.core.storage.sql.S3BinaryManager.AWS_SESSION_TOKEN_PROPERTY;
+import static org.nuxeo.ecm.core.storage.sql.S3BinaryManager.BUCKET_NAME_PROPERTY;
+import static org.nuxeo.ecm.core.storage.sql.S3BinaryManager.BUCKET_PREFIX_PROPERTY;
+import static org.nuxeo.ecm.core.storage.sql.S3BinaryManager.BUCKET_REGION_PROPERTY;
+import static org.nuxeo.ecm.core.storage.sql.S3BinaryManager.SERVERSIDE_ENCRYPTION_PROPERTY;
+
+import java.util.HashMap;
+
+import org.apache.commons.lang3.StringUtils;
+import org.junit.BeforeClass;
+
+import com.amazonaws.SDKGlobalConfiguration;
+
+public class TestS3BinaryManager2WithSSE2AndPolicy extends TestS3BinaryManager {
+
+
+    @BeforeClass
+    public static void beforeClass() {
+
+        String envId = StringUtils.defaultIfBlank(System.getenv(SDKGlobalConfiguration.ACCESS_KEY_ENV_VAR),
+                System.getenv(SDKGlobalConfiguration.ALTERNATE_ACCESS_KEY_ENV_VAR));
+        String envSecret = StringUtils.defaultIfBlank(System.getenv(SDKGlobalConfiguration.SECRET_KEY_ENV_VAR),
+                System.getenv(SDKGlobalConfiguration.ALTERNATE_SECRET_KEY_ENV_VAR));
+        String envToken = StringUtils.defaultIfBlank(System.getenv(SDKGlobalConfiguration.AWS_SESSION_TOKEN_ENV_VAR),
+                "");
+
+        assumeTrue("AWS Credentials not set in the environment variables", StringUtils.isNoneBlank(envId, envSecret));
+
+        PROPERTIES = new HashMap<>();
+        PROPERTIES.put(AWS_ID_PROPERTY, envId);
+        PROPERTIES.put(AWS_SECRET_PROPERTY, envSecret);
+        PROPERTIES.put(AWS_SESSION_TOKEN_PROPERTY, envToken);
+        PROPERTIES.put(BUCKET_NAME_PROPERTY, "nuxeo-s3-directupload-policy");
+        PROPERTIES.put(BUCKET_PREFIX_PROPERTY, "testfolder/");
+        PROPERTIES.put(BUCKET_REGION_PROPERTY, "eu-west-3");
+        // activate SSE
+        PROPERTIES.put(SERVERSIDE_ENCRYPTION_PROPERTY, Boolean.TRUE.toString());
```

Comment:
I can't tell the difference between this and the parent method.  Is it just one line?  Can you just call a parent method?

Reply:
The difference is values put in the PROPERTIES static object, this tests lays on a abstract class used elsewhere and I did not want to change too much of this.
Should I consider a deeper refactor ?

Reply:
I am not keen on duplication, but it these properties are sufficiently different then you should be good as you are.  Its a test so a refactor wouldn't have much risk if you choose that route.

Reply:
Neither am I, but since it's used by other test in azure module that I won't be able to check I prefer not to take that path